src/goLanguageServer.ts

```diff
@@ -281,12 +202,6 @@ export async function registerLanguageFeatures(ctx: vscode.ExtensionContext) {
 			ctx.subscriptions.push(languageServerDisposable);
 		})
 	);
-
-	// gopls is the only language server that provides live diagnostics on type,
-	// so use gotype if it's not enabled.
-	if (!(toolName === 'gopls' && config.features['diagnostics'])) {
-		vscode.workspace.onDidChangeTextDocument(parseLiveFile, null, ctx.subscriptions);
```

Comment:
Wont we need this if user has opted out of diagnostics?

Reply:
I would prefer to avoid these types of fallbacks, as it can make debugging `gopls` issues difficult. I don't see a reason that someone would opt-out of `gopls` diagnostics on-type, but still want diagnostics on type from a different tool.

Reply:
> I don't see a reason that someone would opt-out of gopls diagnostics on-type, but still want diagnostics on type from a different tool.

Previously, it has been mainly due to the issues with the diagnostics feature from gopls. If we are saying that all of those issues are resolved, then my next question would be as to what are the cases where one doesnt want diagnostics on type

Reply:
I think there is some set of people that prefer to have their diagnostics only show up on save, not on-type. I think at this point, if someone is having issues with `gopls` diagnostics, we should work with them to resolve the issue, or they can disable `gopls`, but we can't let people replace the diagnostics - it's too confusing for debugging purposes.